Assets/GameManager.cs

```diff
@@ -1,39 +1,61 @@
 using System.Collections;
 using System.Collections.Generic;
+using Unity.Netcode;
 using UnityEngine;
+using UnityEngine.SceneManagement;
 using UnityEngine.UI;
 
 public class GameManager : MonoBehaviour
 {
-    private Image healthImage;
+    private SharedGameState sharedGameState;
     public int maxHealth = 1;
     [HideInInspector]
     public int currentHealthAmount;
-
+    
     public delegate void HealthChanged(int newHealthValue);
     public event HealthChanged OnHealthChanged = delegate {};
 
+    
     private void Start()
     {
         currentHealthAmount = maxHealth;
+        sharedGameState = GameObject.FindObjectOfType<SharedGameState>();
+
         MinigameHandler.OnPlayerLostMinigame += (int damageAmount) =>
         {
             DepleteHealth(by: damageAmount);
         };
+        
+        OverworldGoal.OnCollidedWithSpaceship += () =>
+        {
+            WonGame();
+        };
     }
 
+  
+
     private void DepleteHealth(int by = 1)
     {
         var damageAmount = by;
         currentHealthAmount -= damageAmount;
         OnHealthChanged(newHealthValue: currentHealthAmount);
+
         if (currentHealthAmount <= 0)
         {
+            
             EndGame();
         }
     }
 
     private void EndGame()
     {
+        sharedGameState.gameOverSceneTransition.Value = true;
+        SceneManager.LoadScene("GameOver");
+    }
+
+    private void WonGame()
```

Comment:
I don't think we need two functions for this. We could probably use only `EndGame` and pass a parameter that decides which scene to transition to or similar. The advantage of that is that we abstract what happens when the game is supposed to end. With two separate functions, you need to check the conditions for winning or loosing a game every single time you want to end it.
Apart from that `WonGame`  is not really a descriptive name.